src/components/posts/postForm.js

```diff
@@ -0,0 +1,158 @@
+import { useState } from "react"
+import { useNavigate } from "react-router-dom"
+import { useEffect } from "react"
+
+export const PostForm = () => {
+
+
+    const navigate = useNavigate()
+    const [categories, setCategories] = useState([])
+    const [post, setNewPost] = useState({
+        user_id: "",
+        category_id: "",
+        title: "",
+        publication_date: "",
+        image_url: "",
+        content: "",
+    })
+
+    useEffect(
+        () => {
+            fetch('http://localhost:8088/categories?_sortBy=label')
+                .then(response => response.json())
+                .then((categoryArray) => {
+                    setCategories(categoryArray)
+                })
+        },
+        []
+    )
+
+
+    const createPost = (postObject) => {
+        return fetch("http://localhost:8088/posts", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(postObject)
+        })
+            .then(response => response.json())
+    }
+
+
+    const saveButton = (event) => {
+        event.preventDefault()
+
+        const localUser = localStorage.getItem("auth_token")
+        const userObject = JSON.parse(localUser)
+
+        const postToSendToAPI = {
+            user_id: userObject,
+            category_id: post.category_id,
+            title: post.title,
+            publication_date: post.publication_date,
+            image_url: post.image_url,
+            content: post.content,
+
+        }
+
+        createPost(postToSendToAPI)
+            .then(newPost => navigate(`/posts/${newPost.id}`))
```

Comment:
i think this should navigate back to /posts, since we don't have a route for a specific post yet